brambling/static/brambling/brambling.utils.js

```diff
@@ -0,0 +1,30 @@
+// Some simple Javascript utilities to be used wherever:
+
+;(function () {
+	"use strict";
+
+	// Filters to be used in Ractive mustaches:
+	brambling.filters = {
+		// Wraps a bit of text in an element:
+		highlight: function (needle, haystack, open_tag, close_tag) {
```

Comment:
Needs to handle the case where needle or haystack is undefined (for whatever reason). Just an `if (!needle || !haystack) return undefined` should do it. Otherwise you get warnings clogging up the console.